internal/provider/build.go

```diff
@@ -45,7 +45,12 @@ func fromImageData(data BuildResourceModel, wd string) (*build.Context, error) {
 	return bc, nil
 }
 
-func doBuild(ctx context.Context, data BuildResourceModel) (v1.Hash, coci.SignedEntity, map[string]string, error) {
+type imagesbom struct {
+	hash string
+	sbom string
```

Comment:
I'd either add a comment to these, or better, change `sbom string` to `content []byte` so it's clearer what kind of data that's going to be.

Reply:
I renamed it to `predicate` and added a `predicate_type` field.